libs/meta/include/meta/type_traits.hpp

```diff
@@ -147,5 +148,45 @@ struct IsNotImplementedImpl
 template <typename A, typename R>
 using IfIsNotImplemented = typename IsNotImplementedImpl<A, R>::Type;
 
+template <class F, class... Args>
+using IsNothrowInvocable =
+    std::integral_constant<bool, noexcept(std::declval<F>()(std::declval<Args>()...))>;
+
+template <class F, class... Args>
+static constexpr auto IsNothrowInvocableV = IsNothrowInvocable<F, Args...>::value;
+
+template <class F, class... Args>
+struct InvokeResult
+{
+  using type = decltype(std::declval<F>()(std::declval<Args>()...));
+};
+
+template <class F, class... Args>
+using InvokeResultT = typename InvokeResult<F, Args...>::type;
+
+template <class C, class RV, class... Args, class... Args1>
+struct InvokeResult<RV (C::*)(Args...), Args1...>
+{
+  using type = RV;
+};
+
+template <class C, class RV, class... Args, class... Args1>
+struct InvokeResult<RV (C::*)(Args...) const, Args1...>
+{
+  using type = RV;
+};
+
+template <class L, class R>
+struct IsSimilar : std::is_same<L, R>
+{
+};
+template <class L, class R>
+static constexpr auto IsSimilarV = IsSimilar<L, R>::value;
+
+template <template <class...> class Ctor, class... LArgs, class... RArgs>
+struct IsSimilar<Ctor<LArgs...>, Ctor<RArgs...>> : std::true_type
```

Comment:
These don't appear to be used in conjunction with this PR. Can we separate them into there own PR if that is the case